### DISCUSSION TITLE
第一次线上会议

### DISCUSSION THREAD

**Discussion by jiacai2050** [zigcc/forum#72] [投票]
# 讨论内容
- 成员间相互介绍，（10min）
- 聊聊社区共建的想法（40min）
  - 视频教程
  - 入门教程
    - https://learnzig.nvimer.org/
    - https://zigcc.github.io/learning-zig/ 
  - 专项主题教程（高级）
    - Zig 开发一个 OS，类似 [Writing an OS in Rust](https://os.phil-opp.com/)
    - Zig 构建系统，如何与 C/C++ 项目结合使用
    - 开发一个 mini LSM
  - 公众号发周报 ，同步 zig 社区最新动态
  - 最佳实践 https://github.com/zigcc/zig-idioms

# 参加要求
- 热爱 Zig
- 有参与其他开源社区维护者优先

最终人数 < 10

# 联系方式
- 微信：jiacai2050，备注：参与 zig 会议

**Comment by chenrui333** [comment]
还可以晚一个小时么，base 在美东

**Reply by jiacai2050** [reply]
加到选项里了

**Reply by chenrui333** [reply]
周五周六 我都可以 (21:00 - 22:00)

**Reply by jiacai2050** [reply]
嗯，不过选择不少了，先不加了。

**Comment by jiacai2050** [comment]
# 2023-12-09 线上会议总结


| 项目         | 参与人员      | 目标                         | 仓库                                                                            |
|------------ |------------- |---------------------------- |------------------------------------------------------------------------------- |
| Zig 写 OS 教程 | 柠檬、西瓜    | 体现 Zig low level 的优势    | [zigcc/how-to-write-os-in-zig](https://github.com/zigcc/how-to-write-os-in-zig) |
| Zig 入门教程 | 金中甲        | 让没有编程背景的人可以有体系的学习 Zig | [learnzig/learnzig](https://github.com/learnzig/learnzig)                       |
| Zig 教学视频 | Onion、Lambert | 同上                |                                                                                 |
| Zig cookbook | 夜白、冯文轩  | 演示如何用 Zig 做某个功能    | [zigcc/zig-cookbook](https://github.com/zigcc/zig-cookbook)                     |
| Zig 构建系统教程 | 贺鹏、陈瑞    | 体验 Zig 编译系统的能力与优势、与其他构建系统的对比 | [zigcc 网站系列文章](https://zigcc.github.io/post/)                             |
| Zig 惯用法   | 全体          | 收集 Zig 编程技巧            | [zigcc/zig-idioms](https://github.com/zigcc/zig-idioms)                         |

## 两周后（2023-12-23），同步进展

**Comment by jiacai2050** [comment]
# 2023-12-20 更新
## zig-os 
- 粗略看完 rust 版本的教程
- zig-os 完成 freestanding 二进制，现在卡在了 bootloader 阶段，相当于第二章的内容 https://os.phil-opp.com/minimal-rust-kernel/

**Comment by jinzhongjia** [comment]
# 2023-12-20 更新
## learnzig
* zig的进阶特性，诸如构建系统、包管理、与C交互均已完成，目前教程内容已基本覆盖日常使用
* 增加了评论区的功能
* 待完成：反射（编译期反射和运行时反射）、内建函数说明（包含使用例子）、未定义行为、wasm、原子操作这些边缘部分
